rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,852 @@
+            # TODO(spolcyn): Replace this with Nibabel's concat_images function
+            # when the dtype issue with save/load cycle is fixed
+            # https://github.com/nipy/nibabel/issues/986
+            newArchiveData = np.concatenate(
+                (archiveData, getNiftiData(incremental.image)), axis=3)
+            newImg = nib.Nifti1Image(newArchiveData,
+                                     affine=archiveImg.affine,
+                                     header=archiveImg.header)
+            newImg.update_header()
+            self._addImage(newImg, imgPath)
+            return True
+
+        # 2.2) Image doesn't exist in archive, but rest of the path is valid for
+        # the archive; create new Nifti file within the archive
+        if self.dirExistsInArchive(dataDirPath) or makePath:
+            logger.debug("Image doesn't exist in archive, creating")
+            self._addImage(incremental.image, imgPath, updateLayout=False)
+            self._addMetadata(incremental.imageMetadata, metadataPath,
+                              updateLayout=False)
+            self._updateLayout()
+            return True
+
+        # 2.3) No image append possible and no creation possible; fail append
+        return False
+
+    @failIfEmpty
+    def getIncremental(self, imageIndex: int = 0, **entities) \
+            -> BidsIncremental:
+        """
+        Creates a BIDS Incremental from the specified part of the archive.
+
+        Args:
+            imageIndex: Index of 3-D image to select in a 4-D image volume.
+            entities: Keyword arguments for entities to filter by. Provide in
+                the format entity='value'.
+
+        Returns:
+            BIDS-Incremental file with the specified image of the archive and
+            its associated metadata.
+
+        Raises:
+            IndexError: If the provided imageIndex goes beyond the bounds of the
+                volume specified in the archive.
+            MissingMetadataError: If the archive lacks the required metadata to
+                make a BIDS Incremental out of an image in the archive.
+            NoMatchError: When no images that match the provided entities are
+                found in the archive
+            RuntimeError:
+                1) When too many images that match the provided entities
+                are found in the archive.
+                2) If the image matching the provided entities has fewer
+                than 3 dimensions or greater than 4.
+
+        Examples:
+            >>> archive = BidsArchive('.')
+            >>> inc = archive.getIncremental(subject='01', task='test')
+            >>> entityFilterDict = {'subject': '01', 'task': 'test'}
+            >>> inc2 = archive.getIncremental(**entityFilterDict)
+            >>> inc == inc2
+            True
+            >>> inc.imageDimensions
+            (64, 64, 27, 1)
+            >>> inc3 = archive.getIncremental(imageIndex=1, **entityFilterDict)
```

Comment:
Should this example be imageIndex=0 to be equal to inc2?

Reply:
The default `imageIndex` is 0, so both calls to `getIncremental` will return the image with index 0.